src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -982,3 +938,29 @@ GetReferencedTableId(Oid foreignKeyId)
 
 	return referencedTableId;
 }
+
+
+/*
+ * Returns true if type of the table with relationId (distributed, reference,
+ * Citus local or local) is included in the flags, false if not
+ */
+static bool
+IsTableTypeIncluded(Oid relationId, int flags)
+{
+	if (IsTableType(relationId, DISTRIBUTED_TABLE))
+	{
+		return flags | INCLUDE_DISTRIBUTED_TABLES;
+	}
+	if (IsTableType(relationId, REFERENCE_TABLE))
+	{
+		return flags | INCLUDE_REFERENCE_TABLES;
+	}
+	if (IsTableType(relationId, CITUS_LOCAL_TABLE))
+	{
+		return flags | INCLUDE_CITUS_LOCAL_TABLES;
+	}
+	if (!IsCitusTable(relationId))
+	{
+		return flags | INCLUDE_LOCAL_TABLES;
+	}
```

Comment:
1) minor: could use else if blocks

2) also, might be good to make this final check the first check, as we usually do in codebase

3) here we assume caller can only pass one of those flags ?
Then I think we should not remove those 3 assertions, but instead, we should extend them.
Also better to mention this somewhere in `typedef enum ExtractForeignKeyConstraintsMode`

Reply:
For 3, we don't assume it anymore. We can include more than one table type and if none of them is included no foreign key is returned.

Reply:
then I guess we should change the structure in this function as it seem to assume only one of them can be passed

Reply:
my bad, you can ignore 3 :)

Reply:
Actually, here we assume the table type can only be one of distributed, reference, Citus local and Postgres local.
So, the code enters one of the `if`s then no matter how many flags there are it only checks the related flag.